ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -59,19 +59,46 @@ let create_as_necessary ~__context ~host =
     let network = get_network_internal ~__context ~self:cluster_ref in
     let (pifref,pifrec) = Xapi_clustering.pif_of_host ~__context network host in
     fix_pif_prerequisites ~__context (pifref,pifrec);
-    Helpers.call_api_functions ~__context (fun rpc session_id ->
-        Client.Client.Cluster_host.create rpc session_id cluster_ref host pifref) |> ignore
+    Helpers.call_api_functions ~__context
+      (fun rpc session_id ->
+        try
+          Client.Client.Cluster_host.create rpc session_id cluster_ref host pifref
+        with Api_errors.(Server_error (_, _)) as err ->
```

Comment:
If a different error surfaces, should this be logged? Or can we ignore them here and let them escape? The same applies below.